src/Microsoft.ML.Recommender/MatrixFactorizationPredictor.cs

```diff
@@ -43,33 +46,42 @@ private static VersionInfo GetVersionInfo()
                 verReadableCur: 0x00010001,
                 verWeCanReadBack: 0x00010001,
                 loaderSignature: LoaderSignature,
-                loaderAssemblyName: typeof(MatrixFactorizationPredictor).Assembly.FullName);
+                loaderAssemblyName: typeof(MatrixFactorizationModelParameters).Assembly.FullName);
         }
         private const uint VersionNoMinCount = 0x00010002;
 
         private readonly IHost _host;
-        // The number of rows.
-        private readonly int _numberOfRows;
-        // The number of columns.
-        private readonly int _numberofColumns;
-        // The rank of the factor matrices.
-        private readonly int _approximationRank;
-        // Packed _numberOfRows by _approximationRank matrix.
-        private readonly float[] _leftFactorMatrix;
-        // Packed _approximationRank by _numberofColumns matrix.
-        private readonly float[] _rightFactorMatrix;
-
-        public PredictionKind PredictionKind
-        {
-            get { return PredictionKind.Recommendation; }
-        }
+        ///<summary> The number of rows.</summary>
+        public readonly int NumberOfRows;
+        ///<summary> The number of columns.</summary>
+        public readonly int NumberOfColumns;
+        ///<summary> The rank of the factor matrices.</summary>
+        public readonly int ApproximationRank;
+        /// <summary>
+        /// Left approximation matrix
+        /// </summary>
+        /// <remarks>
+        /// This is two dimensional matrix with size of <see cref="NumberOfRows"/> * <see cref="ApproximationRank"/> flattened into one-dimensional matrix.
+        /// Row by row.
+        /// </remarks>
+        public readonly IReadOnlyList<float> LeftFactorMatrix;
+        /// <summary>
+        /// Left approximation matrix
+        /// </summary>
+        /// <remarks>
+        /// This is two dimensional matrix with size of <see cref="ApproximationRank"/> * <see cref="NumberOfColumns"/> flattened into one-dimensional matrix.
+        /// Row by row.
+        /// </remarks>
+        public readonly IReadOnlyList<float> RightFactorMatrix;
+
+        public PredictionKind PredictionKind => PredictionKind.Recommendation;
 
-        public ColumnType OutputType { get { return NumberType.Float; } }
+        private ColumnType OutputType => NumberType.Float;
 
-        public ColumnType MatrixColumnIndexType { get; }
-        public ColumnType MatrixRowIndexType { get; }
+        internal ColumnType MatrixColumnIndexType { get; }
+        internal ColumnType MatrixRowIndexType { get; }
```

Comment:
I think in Tom's last internalization PR, he left the methods that return the types of the input/output matrices public.
It could make sense that one wants to access that after, say after having loaded a previously trained model to know what type of data it can score.

Reply:
Tom PR wasn't about INTENALIZE EVERYTHING I CAN SEE, right? 
I ask him and this is his reply:
```
For now probably internal.
Unless you have some scenario that requires it.
```

I don't see such scenarios, @wschin do you see such?

---
In reply to: [255193733](https://github.com/dotnet/machinelearning/pull/2451#discussion_r255193733) [](ancestors = 255193733)

Reply:
Fair enough :) 


---
In reply to: [255211305](https://github.com/dotnet/machinelearning/pull/2451#discussion_r255211305) [](ancestors = 255211305,255193733)

Reply:
Thank you for asking Tom!

---
In reply to: [255211957](https://github.com/dotnet/machinelearning/pull/2451#discussion_r255211957) [](ancestors = 255211957,255211305,255193733)